src/CoreWCF.Primitives/src/CoreWCF/Dispatcher/SecurityServiceDispatcher.cs

```diff
@@ -663,7 +662,9 @@ private void SendFaultIfRequired(Exception e, Message unverifiedMessage, TimeSpa
                 {
                     if (unverifiedMessage.Headers.MessageId != null)
                         faultMessage.InitializeReply(unverifiedMessage);
-                    ((IDuplexChannel)InnerDuplexChannel).SendAsync(faultMessage);
+                    TimeoutHelper timeoutHelper = new TimeoutHelper(timeout);
+                    await ((IDuplexChannel)InnerDuplexChannel).SendAsync(faultMessage);
+                    await ((IDuplexChannel)InnerDuplexChannel).CloseAsync(timeoutHelper.GetCancellationToken());
                 }
```

Comment:
There are some test failures with the error message "Expected dispatcher state to be Opened or Faulted, instead it's Closed", so I think I was mistaken in saying that CloseAsync needs to be added here. The difference between this method and the earlier one is this is for duplex channels so it's not appropriate to close as the channel itself didn't fault, it was just the message was bad.

Reply:
That makes sense. Fixed and squashed the commits in the latest push